src/client/browserContext.ts

```diff
@@ -86,6 +86,10 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel,
       this._serviceWorkers.add(serviceWorker);
       this.emit(Events.BrowserContext.ServiceWorker, serviceWorker);
     });
+    this._channel.on('request', ({ request, page }) => this._onRequest(network.Request.from(request), page));
```

Comment:
I'd do `Page.fromNullable(page)` right here and pass `Page | null` to the function.